discovery/kubernetes/pod.go

```diff
@@ -257,6 +287,26 @@ func (p *Pod) buildPod(pod *apiv1.Pod) *targetgroup.Group {
 	return tg
 }
 
+func attachNodeMetadata(tg *targetgroup.Group, pod *apiv1.Pod, nodeInf cache.SharedInformer) {
+	tg.Labels[nodeNameLabel] = lv(pod.Spec.NodeName)
+
+	obj, exists, err := nodeInf.GetStore().GetByKey(pod.Spec.NodeName)
+	if err != nil {
```

Comment:
Should we emit an info log in this case?

Reply:
The node of the pod not existing in the API suggest something went terribly wrong, I think even an error being logged would be appropriate.